src/org/scada_lts/web/mvc/api/components/cmp/GetHistoryChangesAPI.java

```diff
@@ -0,0 +1,55 @@
+package org.scada_lts.web.mvc.api.components.cmp;
+
+import com.serotonin.mango.Common;
+import com.serotonin.mango.vo.User;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.scada_lts.service.MultiChangesHistoryService;
+import org.scada_lts.service.model.MultiChangeHistoryComponentDTO;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import javax.annotation.Resource;
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * @author grzegorz.bylica@abilit.eu on 17.10.2019
+ */
+@Controller
+public class GetHistoryChangesAPI {
+    private static final Log LOG = LogFactory.getLog(GetHistoryChangesAPI.class);
+
+    @Resource
+    private MultiChangesHistoryService multiChangesHistoryService;
+
+    @RequestMapping(value = "/api/cmp/history/{xIdViewAndIdCmp}", method = RequestMethod.GET)
+    public ResponseEntity<MultiChangeHistoryComponentDTO> getHistory(@PathVariable(name = "xIdViewAndIdCmp") String xidViewAndIdCmp, HttpServletRequest request) {
+        LOG.info("/api/cmp/hisotry xidViewAndIdCmp:" + xidViewAndIdCmp);
+
+        try {
+            User user = Common.getUser(request);
+
+            if (user != null) {
+
+                MultiChangeHistoryComponentDTO mchc = new MultiChangeHistoryComponentDTO(
```

Comment:
Is it possible to change name "mchc"  to more understandable?

Reply:
improved